### ISSUE TITLE
Kotlin 1.1.0 compatibility

### ISSUE THREAD

**Issue by gray419** [nhaarman/mockito-kotlin#156]
Great library, excited to see Kotlin 1.1 compatibility

**Comment by mg6maciej** [comment]
It is 1.1 compatible. The only thing you need to do to use it, if you don't depend on kotlin-reflect in other ways than as a transitive dependency in mockito-kotlin, is to add:
```
    testCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
```
and / or on Android:
```
    androidTestCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
```

**Comment by nhaarman** [comment]
When I test a project that uses `1.1.0` and depends on Mockito-Kotlin with Kotlin `1.0.x`, I only get this warning:

```
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    (...)/kotlin-reflect-1.0.6.jar (version 1.0)
    (...)/kotlin-stdlib-1.1.0.jar (version 1.1)
w: Consider providing an explicit dependency on kotlin-reflect 1.1 to prevent strange errors
```

I am not using instrumentation tests though. As I understand correctly, your build fails if you don't include those lines?

**Comment by mg6maciej** [comment]
Yes, even `./gradlew test` gives this:
```
:app:prepareDebugUnitTestDependencies
Conflict with dependency 'org.jetbrains.kotlin:kotlin-stdlib' in project ':app'. Resolved versions for app (1.1.0) and test app (1.0.6) differ. See http://g.co/androidstudio/app-test-app-conflict for details.
:app:prepareDebugUnitTestDependencies FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:prepareDebugUnitTestDependencies'.
> Dependency Error. See console for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

which is kind of weird, as the tests are run on the computer and not emulator / device.
Same error when running `./gradlew connectedAndroidTest`.

**Comment by nhaarman** [comment]
Ok, so I see the issue here.

In my own project, I only declare the following dependencies:

```
compile kotlinStdLib
testCompile mockitoKotlin
```

The `stdLib` dependency uses `1.1.0`, where `mockitoKotlin` pulls in `1.0.x`.
Since I explicitly depend on `stdlib:1.1.0`, `mockitoKotlin`'s `stdlib` is overridden with `1.1.0`, hence no conflict.
Since `mockitoKotlin` only dependes on `kotlin-reflect:1.0.x`, and my app does not depend on `kotlin-reflect` at all, there is no conflict either.

In your case, your dependencies pull in both `stdlib:1.0.6` and `stdlib:1.1.0` and you don't override them, which is a conflict.
Since `reflect:1.0.6` depends on `reflect:1.0.6`, your get a conflict there as well.

---

**TL;DR**
So the easiest solution is to explicitly depend on the Kotlin version you're using. If you or your dependencies are only using the standard library, the following is enough:

```groovy
compile 'org.jetbrains.kotlin:kotlin-stdlib:1.1.0'
```

If you or other libraries depend on the reflect library as well, add the following:

```
testCompile 'org.jetbrains.kotlin:kotlin-reflect:1.1.0`
```
_Note that I used `testCompile` instead of compile, if you're depending on `kotlin-reflect` for production change this to `compile`._
